### PR TITLE
🐛 Add HIVELOCITY_REGION in the e2e ginkgo code

### LINKED ISSUE #97: CAPHV fails `Missing: ["HIVELOCITY_REGION"]`
/kind bug

```shell
[FAILED] Failed to run clusterctl config cluster
  Unexpected error:
      <*yamlprocessor.errMissingVariables | 0xc000eada40>: 
      value for variables [HIVELOCITY_REGION] is not set. Please set the value using os environment variables or the clusterctl config file
      {
          Missing: ["HIVELOCITY_REGION"],
      }
  occurred
  In [It] at: /home/runner/work/cluster-api-provider-hivelocity/cluster-api-provider-hivelocity/vendor/sigs.k8s.io/cluster-api/test/framework/clusterctl/client.go:292
```

see - https://github.com/hivelocity/cluster-api-provider-hivelocity/actions/runs/5400200656/jobs/9808351567#step:3:599

**What steps did you take and what happened:**
Running the CI.

**What did you expect to happen:**
E2E run successfully

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-hivelocity version:
- Kubernetes version: (use `kubectl version`)
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by aniruddha2000**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #97 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits

### CODE REVIEW

**Comment by batistein** [comment]
@aniruddha2000 did you tried out without pull_request_target?

**Comment by batistein** [comment]
@aniruddha2000 the issue is here: https://github.com/hivelocity/cluster-api-provider-hivelocity/blob/main/test/e2e/config/hivelocity.yaml#L127C1-L127C26

vs.
https://github.com/hivelocity/cluster-api-provider-hivelocity/blob/main/test/e2e/config/hivelocity-ci.yaml#L117-L123

in the ci one this variable is missing. This is why it works locally and why it fails on the ci. We need to make sure that both files are identically. The best would be to remove the need to maintain two files

**Review by batistein — APPROVED** [review]

**Review by batistein — CHANGES_REQUESTED** [review]
Sorry, I didn't realize you deleted the ci one. This one specifies the TAG via the env variable, which is necessary for testing multiple PRs. The operator is built with the tag of the PR number. If possible, you can remove the one for local use and rename the ci one. @guettli Is there anything special in the config for local testing? The default value for TAG is e2e and we just need to change the MANIFEST_PATH to: `../../../` in the default.

**Comment by guettli** [comment]
> Sorry, I didn't realize you deleted the ci one. This one specifies the TAG via the env variable, which is necessary for testing multiple PRs. The operator is built with the tag of the PR number. If possible, you can remove the one for local use and rename the ci one. @guettli Is there anything special in the config for local testing? The default value for TAG is e2e and we just need to change the MANIFEST_PATH to: `../../../` in the default.

@batistein I am unsure what you mean here. I don't think there is something special for local e2e testing.

**Comment by batistein** [comment]
thanks @aniruddha2000 for updating. @guettli Maybe we should talk about this quickly. I think we can facilitate the PR and the e2e config then.

File: test/e2e/data/infrastructure-hivelocity/v1alpha1/cluster-template.yaml [hunk 1]
```diff
@@ -372,6 +373,7 @@ spec:
           provider-id: DEVICE_ID_PREKUBEADM
           read-only-port: "0"
           resolv-conf: /etc/kubernetes/resolv.conf
+          rotate-server-certificates: "true"
```
Comment (batistein) [inline test/e2e/data/infrastructure-hivelocity/v1alpha1/cluster-template.yaml:376]: please move this file into: https://github.com/syself/cluster-api-provider-hetzner/blob/main/.gitignore#L18

**Review by batistein — CHANGES_REQUESTED** [review]

File: test/e2e/data/infrastructure-hivelocity/v1alpha1/cluster-template.yaml (hunk 1 above)
Comment (aniruddha2000) [reply test/e2e/data/infrastructure-hivelocity/v1alpha1/cluster-template.yaml:376]: Done

**Comment by batistein** [comment]
now it looks good. Please squash and change to pull_request_target @aniruddha2000

**Review by batistein — DISMISSED** [review]

**Pull request merged**